test/test_tensordict.py

```diff
@@ -1757,6 +1757,30 @@ def test_split(self, td_name, device, performer):
                                 expected_inner_tensor_size
                             )
 
+    def test_pop(self, td_name, device):
+        td = getattr(self, td_name)(device)
+        assert "a" in td.keys()
+        out = td.pop("a")
+        assert torch.eq(out, getattr(self, td_name)(device)['a'])
+        assert 'a' not in td.keys()
+
+        default = "some value"
+        assert "b" in td.keys()
+        out = td.pop("b", default)
+        assert out != default
```

Comment:
`assert tensor` will fail for every tensor that has more than one element
you should do `assert tensor.all()`

Reply:
I see now:
ValueError: default should be None or a Tensor instance, got some value

Reply:
ah sorry, nevermind
I saw an error which looked like `assert tensor` but it was for the one above that you just corrected :)